samples/terminal-plugin/src/new-terminal-backend.ts

```diff
@@ -0,0 +1,174 @@
+        id: 'show-terminal-with-delay',
+        label: 'Show terminal after 3 sec.'
+    }
+
+    export const DisposeTerminal: theia.Command = {
+        id: 'dispose-terminal',
+        label: 'Dispose terminal after 3 sec.'
+    }
+
+    export const SubscribeToOnDidCloseTerminalEvent: theia.Command = {
+        id: 'subscribe-on-did-close-terminal-event',
+        label: 'Subscribe to onDidCloseTerminal event.'
+    }
+
+    export const SubcribeToOnDidOpenTerminalEvent: theia.Command = {
+        id: 'subscribe-on-did-open-terminal-even',
+        label: 'Subscribe to onDidOpenTerminal event.'
+    }
+
+    export const ShowTerminalList: theia.Command = {
+        id: 'show-terminal-list',
+        label: 'Show list of the terminals'
+    }
+
+    export const ShowActiveTerminal: theia.Command = {
+        id: 'show-active-terminal',
+        label: 'Track current active terminal during 10c.',
+    }
+
+    export const SubscribeToOnDidChangeActiveTerminal: theia.Command = {
+        id: 'subscribe-to-on-did-change-active-terminal',
+        label: 'Subscribe to on did change active terminal event.'
+    }
+}
+
+export function start(context: theia.PluginContext) {
+    context.subscriptions.push(theia.commands.registerCommand(Commands.CreateTerminalWithHelpArgs, () => {
+        const terminal = theia.window.createTerminal('Sh Terminal', 'sh', ['-l']);
+        terminal.show();
+    }));
+
+    context.subscriptions.push(theia.commands.registerCommand(Commands.CreateTerminalWithHelpOptions, () => {
+        const termOptions: theia.TerminalOptions = {
+            name: 'Test terminal',
+            shellPath: '/bin/bash',
+            shellArgs: ['-l'],
+            env: { 'HELLO': 'Hello Theia.' },
+            // cwd: '/home/user/projects/che' // any existed absolute path or url to the folder
+        }
+        const terminal = theia.window.createTerminal(termOptions);
+        terminal.show();
+    }));
+
+    context.subscriptions.push(theia.commands.registerCommand(Commands.SendTextToTheTerminal, () => {
+        const terminal = createTerminalWithOptions();
+        terminal.show();
+        terminal.sendText('clear && echo Theia plugin terminal.\n');
+    }));
+
+    context.subscriptions.push(theia.commands.registerCommand(Commands.HidePanelWithTerminal, () => {
+        const terminal = createTerminalWithOptions();
+        terminal.show();
+        setTimeout(function() {
+            terminal.hide();
+        }, 3000);
+    }));
+
+    context.subscriptions.push(theia.commands.registerCommand(Commands.ShowTerminalWithDelay, () => {
+        const terminal = createTerminalWithOptions();
+        setTimeout(function() {
+            terminal.show();
+        }, 3000);
+    }));
+
+    context.subscriptions.push(theia.commands.registerCommand(Commands.DisposeTerminal, () => {
+        const terminal = createTerminalWithOptions();
+        terminal.show();
+        setTimeout(function() {
```

Comment:
'function() {' -> '()=>' ?
You are using the setTimeout function for several time in code with the same interval. I guess it will be better to create a separate async function sleep. For example:
```
    await sleep(3000);

...
async sleep(milliseconds: number = 0): Promise<Object> {
    return new Promise(resolve => setTimeout(resolve, milliseconds));
}
```